### PR TITLE
Pod workers test

### PR DESCRIPTION
**Pull request by wojtek-t**
Create unit test for pod_workers.go file (on top of #5239)

Requires #5239 to be merged.

### CODE REVIEW

File: pkg/kubelet/pod_workers_test.go [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+	if len(podWorkers.podUpdates) != numPods {
+		t.Errorf("Incorrect number of open channels %v", len(podWorkers.podUpdates))
+	}
+
+	desiredPods := map[types.UID]empty{}
+	desiredPods[types.UID(2)] = empty{}
+	desiredPods[types.UID(14)] = empty{}
+	podWorkers.ForgetNonExistingPodWorkers(desiredPods)
+	if len(podWorkers.podUpdates) != 2 {
```
Comment (vmarmol) [inline pkg/kubelet/pod_workers_test.go]: Can we check that the remaining ones are 2 and 14?

File: pkg/kubelet/pod_workers_test.go [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+		for j := 0; j <= i; j++ {
+			if processed[uid][j] != string(j) {
+				t.Errorf("Pod %v: incorrect order %v, %v", i, j, processed[uid][j])
+			}
+		}
+	}
+}
+
+func TestForgetNonExistingPodWorkers(t *testing.T) {
+	fakeDocker := &dockertools.FakeDockerClient{}
```
Comment (vmarmol) [inline pkg/kubelet/pod_workers_test.go]: Can we use a helper for this? It is very similar to the one above and we may be able to just use the same one.

File: pkg/kubelet/pod_workers_test.go (hunk 1 above)
Comment (wojtek-t) [inline pkg/kubelet/pod_workers_test.go]: Done.

File: pkg/kubelet/pod_workers_test.go (hunk 2 above)
Comment (wojtek-t) [inline pkg/kubelet/pod_workers_test.go]: Done.

**Comment by wojtek-t** [comment]
PTAL

**Comment by vmarmol** [comment]
LGTM, lets get the other PR in then rebase and merge this one.

**Comment by vmarmol** [comment]
Merged the last one, lets rebase :)

**Comment by wojtek-t** [comment]
Rebased.

**Pull request merged**